src/redux/reducers/servers.js

```diff
@@ -1,31 +1,21 @@
-const defaultState = {
-  servers: []
-};
+import uniqBy from 'lodash/uniqBy'
+
+export const defaultState = {
+  servers: [],
+}
 
 const servers = (state = defaultState, action) => {
   switch (action.type) {
     case 'ADD_SERVER':
-      const { servers } = state;
-      const alreadyExists = servers.find(
-        server => server.id === action.server.id
-      );
-
-      if (alreadyExists) {
-        return state;
-      }
-
-      servers.push(action.server);
+      const servers = uniqBy([...state.servers, action.server], 'id')
+      return { ...state, servers }
```

Comment:
This is a weird one. I'm new here, but I assume this is specific to the configuration we attach to `window.config` that tells our client who/how to connect to for data. That config looks something like:

```js
{
  "servers": {
    "dicomWeb": [{ ... }],
    "dimse": [{ ... }]
  },
}
```

The thing is, DIMSE won't be possible with the migration from Meteor to React, and while we can list multiple `dicomWeb` servers, only the first one is currently used, and I've yet to see an `id` key in the config of a server object 🤔 

**Each object does have a `name` field? Perhaps that's a better target for checking uniqueness.**

The second potential issue is using `uniqBy` on that array. Do we know it will prefer the first/existing occurence? If not, you may accidentally replace the original `server` config with the one that has the duplicate id?

Reply:
That's a tough one for me. I was just trying to produce an equivalent code and a corresponding test. Downstream we are using only one server so far. I do not fully understand what this reducer should do.